File: app/src/main/java/com/permissionnanny/simple/RequestSingleUpdateListener.java
package com.permissionnanny.simple;

import android.content.Context;
import android.location.Criteria;
import android.location.Location;
import android.location.LocationListener;
import android.location.LocationManager;
import android.os.Looper;
import com.permissionnanny.ProxyService;
import com.permissionnanny.lib.request.RequestParams;
import com.permissionnanny.lib.request.simple.LocationRequest;

/**
 */
public class RequestSingleUpdateListener extends RequestLocationUpdatesListener {
    public RequestSingleUpdateListener(ProxyService service, String clientAddr) {
        super(service, clientAddr);
    }

    @Override
    public void onLocationChanged(Location location) {
        super.onLocationChanged(location);
        stop();
    }

    /**
     * {@link LocationRequest#requestSingleUpdate(String, LocationListener, Looper)}
     */
    public static class Api extends RequestSingleUpdateListener {
        public Api(ProxyService service, String clientAddr) {
            super(service, clientAddr);
        }

        @Override
        public void register(Context context, RequestParams request) {
            LocationManager lm = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
            lm.requestSingleUpdate(request.string0, this, null);
        }
    }

    /**
     * {@link LocationRequest#requestSingleUpdate(Criteria, LocationListener, Looper)}
     */
    public static class Api1 extends RequestSingleUpdateListener {
        public Api1(ProxyService service, String clientAddr) {
            super(service, clientAddr);
        }

        @Override
        public void register(Context context, RequestParams request) {
            LocationManager lm = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
            lm.requestSingleUpdate(request.criteria0, this, null);
        }
    }
}

File: app/src/main/kotlin/com/permissionnanny/simple/RequestSingleUpdateListener.kt
package com.permissionnanny.simple

import android.content.Context
import android.location.Location
import android.location.LocationManager
import com.permissionnanny.ProxyService
import com.permissionnanny.lib.request.RequestParams
import com.permissionnanny.lib.request.simple.LocationRequest

/**
 */
open class RequestSingleUpdateListener(
        service: ProxyService,
        clientAddr: String)
    : RequestLocationUpdatesListener(service, clientAddr) {

    override fun onLocationChanged(location: Location) {
        super.onLocationChanged(location)
        stop()
    }

    /**
     * [LocationRequest.requestSingleUpdate]
     */
    class Api(service: ProxyService, clientAddr: String) : RequestSingleUpdateListener(service, clientAddr) {

        override fun register(context: Context, request: RequestParams) {
            val lm = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
            lm.requestSingleUpdate(request.string0, this, null)
        }
    }

    /**
     * [LocationRequest.requestSingleUpdate]
     */
    class Api1(service: ProxyService, clientAddr: String) : RequestSingleUpdateListener(service, clientAddr) {

        override fun register(context: Context, request: RequestParams) {
            val lm = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
            lm.requestSingleUpdate(request.criteria0, this, null)
        }
    }
}
